instances/k8smaster/main.tf

```diff
@@ -10,7 +10,14 @@ resource "oci_core_instance" "TFInstanceK8sMaster" {
   hostname_label      = "${var.hostname_label_prefix}-${count.index}"
   image               = "${lookup(data.oci_core_images.ImageOCID.images[0], "id")}"
   shape               = "${var.shape}"
-  subnet_id           = "${var.subnet_id}"
+
+  create_vnic_details {
+    subnet_id        = "${var.subnet_id}"
+    display_name     = "${var.label_prefix}${var.display_name_prefix}-${count.index}"
+    hostname_label   = "${var.hostname_label_prefix}-${count.index}"
+    assign_public_ip = "${(var.control_plane_subnet_access == "private") ? "false" : "true"}"
+    private_ip       = "${var.assign_private_ip == "true" ? cidrhost(lookup(var.network_cidrs,var.subnet_name), count.index+2) : ""}"
```

Comment:
this is static private ip right?

Reply:
Yes, when assign_private_ip is set to true, this will be a static private ip.